regression/cuda/test_float_all_ops.c

```diff
@@ -0,0 +1,437 @@
+/******************************************************************************/
+/* A[N] [op] B[N] --> C[N]                                                    */
+/* Runs all floating point operations on a vector of input data               */
+/* Operations include <add, sub, mul, div, compare, convert>                  */
+/* Includes testing of corner cases (+- inf, div by zero, etc.)               */
+/* Runs on a 1x1 grid of 2x2 tile group.                                      */
+/* Grid dimensions are prefixed at 1x1. --> block_size_x is set to N.         */
+/* This tests uses the software/spmd/bsg_cuda_lite_runtime/float_all_ops/     */
+/* manycore binary in the BSG Manycore repository.                            */
+/******************************************************************************/
+
+
+#include "test_float_all_ops.h"
+//#include <math.h>
+
+#define TEST_NAME "test_float_all_ops"
+#define ALLOC_NAME "default_allocator"
+
+#define NUM_OPS 6
+#define NUM_CORNER_CASES 6
+
+char *operations[NUM_OPS] = {"ADD", "SUB", "MUL", "DIV", "CMPR", "CVRT"};
+
+void host_float_all_ops (float *A, float *B,
+                         float *res_add, float *res_sub,
+                         float *res_mul, float *res_div,
+                         float *res_compare, float *res_convert,
+                         int N){
+	for (int i = 0; i < N; i ++) { 
+		res_add[i] = A[i] + B[i];
+		res_sub[i] = A[i] - B[i];
+		res_mul[i] = A[i] * B[i];
+		res_div[i] = A[i] / B[i];
+		res_compare[i] = (A[i] >= B[i] ? 1 : 0);
```

Comment:
Could this generate all possible instructions (FLT, FLE, FEQ)?

Reply:
That's actually a good point, thanks! I'm not sure though, is there a way to ensure it generates all these instructions? Or is there a way to call them specifically on the RISCV?

Reply:
Small error on my part, I realize this is host code, this comment actually goes for the kernel code. Nevertheless, seeing the disassembly is the only way I believe, but that's not practical. So, I think, in kernel code, ops could actually be replaced by precise inline assembly functions with instructions we want to exercise. What do you think?